currency_rate_update_TH_BOT/tests/test_currency_rate_update_TH_BOT.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2023 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import datetime
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields
+from odoo.exceptions import UserError
+from odoo.tests import common
+
+
+class TestResCurrencyRateProviderBOT(common.TransactionCase):
+    def setUp(self):
```

Comment:
```suggestion
    def setUp(self):
```
Please change it to setupclass